Test case to verify concurrent deletion and creation of multiple PVCs

jilju opened the pull request:
Signed-off-by: Jilju Joy <jijoy@redhat.com>

jilju commented:
Test case log:
```
$ python3.7 -m run_ocsci tests/test_pvc_concurrent_deletion_creation.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.3, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /home/jijoy/python3-venv-ocs-ci/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/jijoy/OCS-CI/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.0.5
collected 1 item                                                                                                                                                                                                  

tests/test_pvc_concurrent_deletion_creation.py::TestMultiplePvcConcurrentDeletionCreation::test_multiple_pvc_concurrent_creation_deletion 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
02:00:49 - MainThread - root - INFO - All logs are located under /tmp/logs
02:00:49 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
02:00:52 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-njvvb ceph auth get-key client.admin --format json-pretty
02:00:57 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-1202004931
02:00:57 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/Secretp8rz77ou -o yaml
02:00:58 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-1202004931 -o yaml
02:01:00 - MainThread - resources.ocs - INFO - Adding CephBlockPool with name cbp-test-1202010009
02:01:00 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/CephBlockPool5s7nbksh -o yaml
02:01:01 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-1202010009 -o yaml
02:01:02 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-1202010009 exists
02:01:02 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
02:01:04 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-njvvb ceph osd lspools --format json-pretty
02:01:08 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-1202010875
02:01:08 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/StorageClassjk5klik2 -o yaml
02:01:10 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-1202010875 -o yaml
02:01:11 - MainThread - utility.utils - INFO - Executing command: oc new-project test-project
02:01:13 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Creating Persistent Volume Claim test-pvc1
02:01:13 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc1
02:01:13 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim60zppe4k -o yaml
02:01:15 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
02:01:16 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Created Persistent Volume Claim test-pvc1
02:01:16 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Creating Persistent Volume Claim test-pvc2
```
Skipping because the log in too long.

```
02:18:09 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Successfully deleted initial 100 PVCs
02:18:09 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
02:18:10 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-njvvb rbd ls -p cbp-test-1202010009 --format json
02:18:14 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Verified: PVs associated with deleted PVCs are also deleted
PASSED                                                                                                                                                                                                      [100%]
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
02:18:14 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Executing command: for i in `seq 1 100`;do oc delete pvc test-pvc-re$i -n test-project;done
02:21:05 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Newly created 100 PVCs are now deleted.
02:21:05 - MainThread - utility.utils - INFO - Executing command: oc project openshift-storage
02:21:07 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Project test-project
02:21:09 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-1202010875 -o yaml
02:21:11 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete StorageClass storageclass-test-1202010875
02:21:13 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-1202010009 -o yaml
02:21:14 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete CephBlockPool cbp-test-1202010009
02:21:16 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-1202004931 -o yaml
02:21:17 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Secret secret-test-1202004931
```

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+    Delete project
+    """
+    # Delete newly created PVCs
+    command = (
```
create a function for async_delete_pvcs() under pvc.py as helper function

jilju replied:
To delete PVCs in async way, run_async method in utils.py can be used. I think the same approach is appropriate in this test case.

vasukulkarni replied:
we have PVC class, I think the functionality of execute command can be moved inside that class if it doesn't already exists there.

vasukulkarni replied:
will take this up as follow on PR

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+    log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
+
+    # Switch to default project
+    run_cmd(f'oc project {defaults.ROOK_CLUSTER_NAMESPACE}')
```
please add this to ocp.py as a function

jilju replied:
PR #224

jilju replied:
Done. PR 224 is merged.

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+    PROJECT.delete(resource_name=TEST_PROJECT)
+
+
+def create_multiple_pvc(pvc_base_name, number_of_pvc, sc_name):
```
move to pvc.py as a helper function.
if you can also add an option to create in async way it will be great

jilju replied:
Moved to pvc.py PR #236
Creating in async way can be done by using run_async in utils.py by passing a command to create more than one pvc. Here the implementation is using PVC class. So creating in async way is not implemented in PR 236.

jilju replied:
Done. PR 236 is merged.

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+        log.info(f'Created Persistent Volume Claim {pvc_name}')
+
+
+def run_async(command):
```
move to utils.py
Alos, if you can modify that to get a function as input and run it in parallel it will be more pythonic

jilju replied:
PR #227
I think, running  method in parallel will be a different scenario and can be achieved using multithreading. I like to keep this one separate from that kind of scenario. So that is not implemented in PR 227

jilju replied:
Done. PR 227 is merged.

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+    pvc_objs_initial = []
+    pvc_objs_new = []
+
+    @tier1
```
please move this to decorate the class

jilju replied:
Done

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,215 @@
+        log.info(f'Created {self.number_of_pvc} new PVCs.')
+        self.pvc_objs_new = PVC_OBJS[:]
+
+        # Verify PVCs are Bound
```
helper function under pvc.py

jilju replied:
Here 'status' attribute of PVC object is used to verify the status. I think this is feasible based on the way this test case is implemented.

RazTamir commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -82,61 +83,15 @@ def teardown(self):
-    run_cmd(f'oc project {defaults.ROOK_CLUSTER_NAMESPACE}')
+    ret = ocp.switch_to_default_rook_cluster_project()
+    if not ret:
+        log.error('Failed to switch to default project')
```
and assert?

jilju replied:
Done

vasukulkarni commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,174 @@
+        """
+        # Start deleting 100 PVCs
+        command = (
+            f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
```
are you using oc command directly here?  we have pvc class now, can you move the functionality there?

jilju replied:
@vasukulkarni
The intention here is to issue a command for deleting PVCs and immediately start creating another set of PVCs so that both process will run simultaneously. The command to delete PVCs will be verified in the last part of the test case.

vasukulkarni commented on tests/test_pvc_concurrent_deletion_creation.py:
```diff
@@ -0,0 +1,174 @@
+
+        # Create 100 new PVCs
+        # Parameters for PVC yaml as dict
+        pvc_data = defaults.CSI_PVC_DICT.copy()
```
do you want deepcopy() here?

jilju replied:
Done

ebenahar commented:
@jilju as there were a few changes recently, please re-verify and paste the log. Thanks

jilju commented:
Test case log (used 5 PVCs instead of 100 PVCs to reduce logs and for  easy analysis. Test case is also verified using 100 PVCs.) 

```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/jijoy/OCS-CI/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.0.5
collected 1 item                                                                                                                                                                                                  

tests/test_pvc_concurrent_deletion_creation.py::TestMultiplePvcConcurrentDeletionCreation::test_multiple_pvc_concurrent_creation_deletion 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
23:21:01 - MainThread - root - INFO - All logs are located under /tmp/logs
23:21:01 - MainThread - tests.conftest - INFO - Running OCS basic installation
23:21:01 - MainThread - oc.openshift_ops - INFO - Testing access to cluster with /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig
23:21:01 - MainThread - utility.utils - INFO - Executing command: oc cluster-info
23:21:03 - MainThread - oc.openshift_ops - INFO - Access to cluster is OK!
23:21:03 - MainThread - tests.conftest - INFO - The installation is skipped because the cluster is running
23:21:03 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod -A -o yaml
23:21:16 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass -A -o yaml
23:21:18 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephFileSystem -A -o yaml
23:21:20 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool -A -o yaml
23:21:22 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolume -A -o yaml
23:21:24 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
23:21:26 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret -A -o yaml
23:21:56 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Namespace -A -o yaml
23:21:58 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
23:22:00 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 ceph auth get-key client.admin --format json-pretty
23:22:08 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-2423220841
23:22:08 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/Secretjm8oiadc -o yaml
23:22:10 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-2423220841 -o yaml
23:22:11 - MainThread - resources.ocs - INFO - Adding CephBlockPool with name cbp-test-2423221170
23:22:11 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/CephBlockPoolssp4p2ux -o yaml
23:22:13 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-2423221170 -o yaml
23:22:15 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-2423221170 exists
23:22:15 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
23:22:17 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 ceph osd lspools --format json-pretty
23:22:25 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-2423222582
23:22:25 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/StorageClass33aw3ldh -o yaml
23:22:27 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-2423222582 -o yaml
23:22:29 - MainThread - utility.utils - INFO - Executing command: oc new-project test-project
23:22:32 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc1
23:22:32 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc1
23:22:32 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim2k4wu978 -o yaml
23:22:34 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
23:22:36 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc1
23:22:36 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc2
23:22:36 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc2
23:22:36 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim6nvm5ilp -o yaml
23:22:38 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
23:22:41 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc2
23:22:41 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc3
23:22:41 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc3
23:22:41 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim5gpchope -o yaml
23:22:42 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
23:22:44 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc3
23:22:44 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc4
23:22:44 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc4
23:22:44 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimna8t03j8 -o yaml
23:22:45 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
23:22:47 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc4
23:22:47 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc5
23:22:47 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc5
23:22:47 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimo_mivk6p -o yaml
23:22:49 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
23:22:52 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc5
23:22:52 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Created initial 5 PVCs
23:22:52 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
23:22:54 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
23:22:56 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
23:22:58 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
23:23:00 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
23:23:02 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Initial 5 PVCs are in Bound state
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
23:23:02 - MainThread - utility.utils - INFO - Executing command: for i in `seq 1 5`;do oc delete pvc test-pvc$i -n test-project;done
23:23:02 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re1
23:23:02 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re1
23:23:02 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim9po8tumh -o yaml
23:23:03 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re1 -o yaml
23:23:05 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re1
23:23:05 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re2
23:23:05 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re2
23:23:05 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim20oh5kh9 -o yaml
23:23:07 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re2 -o yaml
23:23:09 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re2
23:23:09 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re3
23:23:09 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re3
23:23:09 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimg4jbhz4d -o yaml
23:23:11 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re3 -o yaml
23:23:12 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re3
23:23:12 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re4
23:23:12 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re4
23:23:12 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimf94wxced -o yaml
23:23:14 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re4 -o yaml
23:23:16 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re4
23:23:16 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re5
23:23:16 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re5
23:23:16 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimfns_zq96 -o yaml
23:23:17 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re5 -o yaml
23:23:19 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re5
23:23:19 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Created 5 new PVCs.
23:23:19 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re1 -o yaml
23:23:21 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re2 -o yaml
23:23:23 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re3 -o yaml
23:23:24 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re4 -o yaml
23:23:26 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re5 -o yaml
23:23:27 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Verified: Newly created PVCs are in Bound state.
23:23:27 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Return values of command: for i in `seq 1 5`;do oc delete pvc test-pvc$i -n test-project;done.
                                                                             retcode:0
                                                                             stdout:persistentvolumeclaim "test-pvc1" deleted
                                                                             persistentvolumeclaim "test-pvc2" deleted
                                                                             persistentvolumeclaim "test-pvc3" deleted
                                                                             persistentvolumeclaim "test-pvc4" deleted
                                                                             persistentvolumeclaim "test-pvc5" deleted
                                                                             
                                                                             stderr:None
23:23:27 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
23:23:30 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc1" not found
23:23:30 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc1 does not exists 
23:23:30 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
23:23:32 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc2" not found
23:23:32 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc2 does not exists 
23:23:32 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
23:23:34 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc3" not found
23:23:34 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc3 does not exists 
23:23:34 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
23:23:36 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc4" not found
23:23:36 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc4 does not exists 
23:23:36 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
23:23:38 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc5" not found
23:23:38 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc5 does not exists 
23:23:38 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Successfully deleted initial 5 PVCs
23:23:38 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
23:23:41 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 rbd ls -p cbp-test-2423221170 --format json
23:23:46 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Verified: PVs associated with deleted PVCs are also deleted
PASSED                                                                                                                                                                                                      [100%]
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
23:23:46 - MainThread - utility.utils - INFO - Executing command: for i in `seq 1 5`;do oc delete pvc test-pvc-re$i -n test-project;done
23:23:58 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Newly created 5 PVCs are now deleted.
23:23:58 - MainThread - ocs.ocp - INFO - Switching to project openshift-storage
23:23:58 - MainThread - utility.utils - INFO - Executing command: oc project openshift-storage
23:24:01 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Project test-project
23:24:04 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-2423222582 -o yaml
23:24:05 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete StorageClass storageclass-test-2423222582
23:24:08 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-2423221170 -o yaml
23:24:09 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete CephBlockPool cbp-test-2423221170
23:24:12 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-2423220841 -o yaml
23:24:13 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Secret secret-test-2423220841
23:24:16 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod -A -o yaml
23:24:28 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass -A -o yaml
23:24:30 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephFileSystem -A -o yaml
23:24:32 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool -A -o yaml
23:24:34 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolume -A -o yaml
23:24:37 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
23:24:38 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret -A -o yaml
23:25:09 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Namespace -A -o yaml

```

jilju commented:
> @jilju as there were a few changes recently, please re-verify and paste the log. Thanks

@ebenahar Done

jilju commented:
Test case log (used 5 PVCs instead of 100 PVCs to reduce logs and for easy analysis. Test case is also verified using 100 PVCs.)
```
tests/test_pvc_concurrent_deletion_creation.py::TestMultiplePvcConcurrentDeletionCreation::test_multiple_pvc_concurrent_creation_deletion 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
14:57:50 - MainThread - root - INFO - All logs are located under /tmp/logs
14:57:50 - MainThread - tests.conftest - INFO - Running OCS basic installation
14:57:50 - MainThread - oc.openshift_ops - INFO - Testing access to cluster with /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig
14:57:50 - MainThread - utility.utils - INFO - Executing command: oc cluster-info
14:57:52 - MainThread - oc.openshift_ops - INFO - Access to cluster is OK!
14:57:52 - MainThread - tests.conftest - INFO - The installation is skipped because the cluster is running
14:57:52 - Dummy-1 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod -A -o yaml
14:57:52 - Dummy-2 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass -A -o yaml
14:57:52 - Dummy-3 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephFileSystem -A -o yaml
14:57:52 - Dummy-4 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool -A -o yaml
14:57:52 - Dummy-5 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolume -A -o yaml
14:57:52 - Dummy-6 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
14:57:52 - Dummy-7 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret -A -o yaml
14:57:52 - Dummy-8 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Namespace -A -o yaml
14:58:41 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:58:43 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 ceph auth get-key client.admin --format json-pretty
14:58:48 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-2514584806
14:58:48 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/Secret9uqumyn8 -o yaml
14:58:49 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-2514584806 -o yaml
14:58:51 - MainThread - resources.ocs - INFO - Adding CephBlockPool with name cbp-test-2514585122
14:58:51 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/CephBlockPooln4te91s8 -o yaml
14:58:52 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-2514585122 -o yaml
14:58:54 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-2514585122 exists
14:58:54 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:58:55 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 ceph osd lspools --format json-pretty
14:59:05 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-2514590540
14:59:05 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/StorageClass8zb_42cy -o yaml
14:59:06 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-2514590540 -o yaml
14:59:08 - MainThread - utility.utils - INFO - Executing command: oc new-project test-project
14:59:10 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc1
14:59:10 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc1
14:59:10 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimrivdypiv -o yaml
14:59:12 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
14:59:13 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc1
14:59:13 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc2
14:59:13 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc2
14:59:13 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimeikuto4f -o yaml
14:59:15 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
14:59:17 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc2
14:59:17 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc3
14:59:17 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc3
14:59:17 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim_88o3e5y -o yaml
14:59:18 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
14:59:25 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc3
14:59:25 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc4
14:59:25 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc4
14:59:25 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimlmul7jul -o yaml
14:59:27 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
14:59:28 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc4
14:59:28 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc5
14:59:28 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc5
14:59:28 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimhqp6xhz0 -o yaml
14:59:30 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
14:59:31 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc5
14:59:31 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Created initial 5 PVCs
14:59:31 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
14:59:38 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
14:59:40 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
14:59:41 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
14:59:42 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
14:59:44 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Initial 5 PVCs are in Bound state
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
14:59:44 - MainThread - utility.utils - INFO - Executing command: for i in `seq 1 5`;do oc delete pvc test-pvc$i -n test-project;done
14:59:44 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re1
14:59:44 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re1
14:59:44 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimv0tk6sfv -o yaml
14:59:46 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re1 -o yaml
14:59:47 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re1
14:59:47 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re2
14:59:47 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re2
14:59:47 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimqwgcgt06 -o yaml
14:59:49 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re2 -o yaml
14:59:50 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re2
14:59:50 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re3
14:59:50 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re3
14:59:50 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimn0b1grtk -o yaml
14:59:52 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re3 -o yaml
14:59:53 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re3
14:59:53 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re4
14:59:53 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re4
14:59:53 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaimh8j80ek6 -o yaml
14:59:55 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re4 -o yaml
14:59:56 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re4
14:59:56 - MainThread - resources.pvc - INFO - Creating Persistent Volume Claim test-pvc-re5
14:59:56 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name test-pvc-re5
14:59:56 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig create -f /tmp/PersistentVolumeClaim4dfff7ji -o yaml
14:59:58 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re5 -o yaml
14:59:59 - MainThread - resources.pvc - INFO - Created Persistent Volume Claim test-pvc-re5
14:59:59 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Created 5 new PVCs.
14:59:59 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re1 -o yaml
15:00:01 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re2 -o yaml
15:00:02 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re3 -o yaml
15:00:04 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re4 -o yaml
15:00:05 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc-re5 -o yaml
15:00:07 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Verified: Newly created PVCs are in Bound state.
15:00:07 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Return values of command: for i in `seq 1 5`;do oc delete pvc test-pvc$i -n test-project;done.
                                                                             retcode:0
                                                                             stdout:persistentvolumeclaim "test-pvc1" deleted
                                                                             persistentvolumeclaim "test-pvc2" deleted
                                                                             persistentvolumeclaim "test-pvc3" deleted
                                                                             persistentvolumeclaim "test-pvc4" deleted
                                                                             persistentvolumeclaim "test-pvc5" deleted
                                                                             
                                                                             stderr:None
15:00:07 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc1 -o yaml
15:00:08 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc1" not found
15:00:08 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc1 does not exists 
15:00:08 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc2 -o yaml
15:00:10 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc2" not found
15:00:10 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc2 does not exists 
15:00:10 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc3 -o yaml
15:00:12 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc3" not found
15:00:12 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc3 does not exists 
15:00:12 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc4 -o yaml
15:00:14 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc4" not found
15:00:14 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc4 does not exists 
15:00:14 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim test-pvc5 -o yaml
15:00:21 - MainThread - utility.utils - ERROR - CMD error:: Error from server (NotFound): persistentvolumeclaims "test-pvc5" not found
15:00:21 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Expected: PVC test-pvc5 does not exists 
15:00:21 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Successfully deleted initial 5 PVCs
15:00:21 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
15:00:22 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-jvhd6 rbd ls -p cbp-test-2514585122 --format json
15:00:26 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Verified: PVs associated with deleted PVCs are also deleted
PASSED                                                                                                                                                                                                      [100%]
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
15:00:26 - MainThread - utility.utils - INFO - Executing command: for i in `seq 1 5`;do oc delete pvc test-pvc-re$i -n test-project;done
15:00:35 - MainThread - tests.test_pvc_concurrent_deletion_creation - INFO - Newly created 5 PVCs are now deleted.
15:00:35 - MainThread - ocs.ocp - INFO - Switching to project openshift-storage
15:00:35 - MainThread - utility.utils - INFO - Executing command: oc project openshift-storage
15:00:38 - MainThread - utility.utils - INFO - Executing command: oc -n test-project --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Project test-project
15:00:40 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass storageclass-test-2514590540 -o yaml
15:00:42 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete StorageClass storageclass-test-2514590540
15:00:43 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool cbp-test-2514585122 -o yaml
15:00:45 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete CephBlockPool cbp-test-2514585122
15:00:47 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret secret-test-2514584806 -o yaml
15:00:48 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig delete Secret secret-test-2514584806
15:00:50 - Dummy-1 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Pod -A -o yaml
15:00:50 - Dummy-8 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get StorageClass -A -o yaml
15:00:50 - Dummy-7 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephFileSystem -A -o yaml
15:00:50 - Dummy-2 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get CephBlockPool -A -o yaml
15:00:50 - Dummy-9 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolume -A -o yaml
15:00:50 - Dummy-10 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
15:00:50 - Dummy-11 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Secret -A -o yaml
15:00:50 - Dummy-12 - utility.utils - INFO - Executing command: oc --kubeconfig /home/jijoy/jijoy-ocs-deployment/auth/kubeconfig get Namespace -A -o yaml


=========================================================================================== 1 passed in 222.27 seconds ============================================================================================
```
